### PR TITLE
WIP get travis green

### PR DESCRIPTION
**Pull request by jrochkind**
With no code changes in this repo, travis on master went from green to red between:

* https://travis-ci.org/samvera/sufia/builds/324697416 (pass)
* https://travis-ci.org/samvera/sufia/builds/327356744 (fail)

So the change is likely in a new version of some dependency(ies) allowed by spec. 

unclear why only tests under rails 5 fail and not under rails 4.2. 

Investigating to get green, with as few changes as possible. In some cases it
may be acceptable to lock to older dependency instead of allowing new that breaks.
In some cases, a minor release of an upstream dependency may be introducing backwards-break change, which should be reported.

### CODE REVIEW

**Comment by jrochkind** [comment]
On travis, it's trying to download solr 6.5.0 -- I'm not really sure _why_ it's doing this, what configuration there says to download that solr instead of the latest. I can't find anything. 

On my local computer if I run tests, I get latest solr 7. 

But anyway, on travis trying to download solr 6.5.0, it's being rate limited by apache unfortunately. So we've got to wait until tomorrow to see if tests pass. :(

**Comment by jrochkind** [comment]
So there is the problem everyone in other samvera projects is having with trying to download a solr from apache and getting rate limited.  I am getting this to be green by just waiting for a new quota period and manually rebuilding. 

It will likely cause problems in future travis builds, but it's really unclear how to fix it. It probably requires changes to the `solr_wrapper_test.yml` file, whch in this case is generated by an `active_fedora` generator, that is called by a `hydra-head` generator, that is (I think? might be more intervening things) called by a `curation_concerns` generator, that is called by the sufia `install` generator, that is called by the engine_cart set up test app generator. Phew. Those layers really don't make things easy. 

I guess the hacky way to fix it would be to just have the sufia test app generator generate it's own `solr_wrapper_test.yml` _on top of_ what AF is putting there, with desired parameters to get around the apache quota (either download from princeton mirror, or use travis caching, or both).  I honestly can't think of a better way. I think it's out of scope for this particular issue/PR though.

**Comment by RudyOnRails** [comment]
@jrochkind just curious, could you share the travis line for:

> On travis, it's trying to download solr 6.5.0 -- I'm not really sure why it's doing this, what configuration there says to download that solr instead of the latest. I can't find anything.

**Comment by jrochkind** [comment]
@RudyOnRails Ah, thanks for the sanity check. It looks like that _isn't_ currently the problem. 

Not sure what the problem currently is. I'll try rebuilding travis again, might just be flaky tests. 

The apache rate limit problem _will_ come back though. Let me see if I can find the line from previous travis builds that led me to say the build on travis is (for mysterious reasons) trying to use solr 6.5.0....   alas, it seems to be gone from history, perhaps when you travis `rebuild` (rather than a new build) it doesn't keep the log from the previous run of that build. Well, we'll see if it comes back. 

And I'll try a simple travis rebuild again to see if it goes green this time... or gets the apache rate limit again.

**Comment by RudyOnRails** [comment]
@jrochkind yeah, I peeked at that failing build and there was only 1 failure that had to do with a click event that was being called on something that was nil/notfound. If that pops up, we can look into inserting a sleep or something.

File: lib/sufia.rb [hunk 1]
```diff
@@ -20,6 +20,8 @@
 require 'sufia/inflections'
 require 'kaminari_route_prefix'
 
+require 'sufia/hydra_derivatives_compat'
+
```
Comment (RudyOnRails) [inline lib/sufia.rb]: I think once https://github.com/samvera/hydra-derivatives/pull/182 is merged and 3.4.1 is released on rubygems, you can swap this with a `bundle update hydra-derivatives`

File: lib/sufia/hydra_derivatives_compat.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+   (hydra_derivatives_gem.version.release < Gem::Version.new('3.5')) &&
+   !Hydra::Derivatives::IoDecorator.instance_methods.include?(:original_name=)
+
+  # rubocop:disable Alias
+  Hydra::Derivatives::IoDecorator.class_eval do
+    alias original_name= original_filename=
+    deprecation_deprecate :"original_name=" => 'original_name= has been deprecated. Use original_filename= instead. This will be removed in hydra-derivatives 4.0'
+  end
+  # rubocop:enable Alias
+end
```
Comment (RudyOnRails) [inline lib/sufia/hydra_derivatives_compat.rb]: And this whole file can go obviously, after you get the fix from the HD gem...

File: solr/config/schema.xml [hunk 3]
```diff
@@ -329,17 +329,11 @@
...
+ <!-- Field to use to determine and enforce document uniqueness.
       Unless this field is marked with required="false", it will be a required field
    -->
  <uniqueKey>id</uniqueKey>
 
- <!-- field for the QueryParser to use when an explicit fieldname is absent -->
- <!--  <defaultSearchField>text</defaultSearchField> -->
-
- <!-- SolrQueryParser configuration: defaultOperator="AND|OR" -->
- <solrQueryParser defaultOperator="OR"/>
```
Comment (RudyOnRails) [inline solr/config/schema.xml]: I'm a bit confused, sorry Jon, why are we removing this?

File: solr/config/schema.xml [hunk 4]
```diff
@@ -59,32 +59,32 @@
 
   <types>
     <fieldType name="string" class="solr.StrField" sortMissingLast="true" />
-    <fieldType name="boolean" class="solr.BoolField" sortMissingLast="true"/>    
+    <fieldType name="boolean" class="solr.BoolField" sortMissingLast="true"/>
```
Comment (RudyOnRails) [inline solr/config/schema.xml]: I assume 99% of these diffs are whitespace errors, minus the `<solrQueryParser defaultOperator="OR"/>`?

**Comment by jrochkind** [comment]
Okay, with the release of hydra-derivatives 3.4.1, sufia is now green on _master_, _without_ this PR. 

So we could just leave it be. 

* sufia _may_ not work with hydra-derivatives 3.4.0. Not sure if it was just a test environment failure, or if actual sufia will have problems. But either way it's probably not a disaster, bugs happen, there are bugs in 3.4.0, if you encounter this one you upgrade to 3.4.1 and you're good. 

* Sufia will not currently work (with default sufia solr config) with Solr 7.  I am _not sure_ why travis is running with solr 6, I can't find any logic that is pinning solr instead of using latest one, but travis builds _are_ happening with solr 6.5.0, and not testing under Solr 7.  But sufia is currently in intentional life support, we aren't neccesarily trying to allow people to use a newer version of Solr than the one that was latest when sufia 7 was published. 

Any thoughts on any of this? @mjgiarlo @RudyOnRails ? Anyone else anyone knows using sufia 7?

**Comment by jrochkind** [comment]
Okay, since the build is passing on master now, without this PR, I'm just going to close this one. 

Not sure why the build is being done with sufia 6 and not 7, it wouldn't pass on sufia 7, but good enough for now.

**Pull request closed without merging**